### PR TITLE
Add test for SIGTERM functionality

### PR DESCRIPTION
**Pull request by pakrym**
@Tratcher

### CODE REVIEW

File: test/Microsoft.AspNetCore.Hosting.TestSites/project.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+  "version": "1.1.0-*",
+  "dependencies": {
+    "Microsoft.AspNetCore.Server.Kestrel": "1.1.0-*",
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.TestSites/project.json]: Circular dependency

File: src/Microsoft.AspNetCore.Server.IntegrationTesting/xunit/TestProjectHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                        return projectFileInfo.DirectoryName;
+                    }
+
+                    directoryInfo = directoryInfo.Parent;
+                }
+                while (directoryInfo.Parent != null);
+
+                throw new Exception($"Project root could not be found using {applicationBasePath}");
+            }
+
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting/xunit/TestProjectHelpers.cs]: nit: newline

File: test/Microsoft.AspNetCore.Hosting.FunctionalTests/Microsoft.AspNetCore.Hosting.FunctionalTests.xproj [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0.25420" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">14.0.25420</VisualStudioVersion>
+    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
+  </PropertyGroup>
+  <Import Project="$(VSToolsPath)\DotNet\Microsoft.DotNet.Props" Condition="'$(VSToolsPath)' != ''" />
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>fc578f4e-171c-4f82-b301-3abf6318d082</ProjectGuid>
+    <RootNamespace>Microsoft.AspNetCore.Hosting.FunctionalTests</RootNamespace>
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.Hosting.FunctionalTests/Microsoft.AspNetCore.Hosting.FunctionalTests.xproj]: Run Dougs script to clean this up

File: test/Microsoft.AspNetCore.Hosting.FunctionalTests/ShutdownTests.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                ApplicationType = ApplicationType.Portable,
+                PublishApplicationBeforeDeployment = true
+            };
+
+            using (var deployer = new SelfHostDeployer(deploymentParameters, logger))
+            {
+                deployer.Deploy();
+
+                // Wait for application to start
+                System.Threading.Thread.Sleep(1000);
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.FunctionalTests/ShutdownTests.cs]: can you query the HostProcess to see if it's started? Why was the delay needed?

File: test/Microsoft.AspNetCore.Hosting.FunctionalTests/ShutdownTests.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+                string output = string.Empty;
+                deployer.HostProcess.OutputDataReceived += (sender, args) => output += args.Data + '\n';
+
+                SendSIGINT(deployer.HostProcess.Id);
+
+                deployer.HostProcess.WaitForExit();
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.FunctionalTests/ShutdownTests.cs]: Is this going to make the test hang if something goes wrong? Any way to timeout here?

File: test/Microsoft.AspNetCore.Hosting.TestSites/Program.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        public void Dispose()
+        {
+            _serverEvent.Wait();
+        }
+
+        public IFeatureCollection Features { get; } = new FeatureCollection();
+
+        public void Start<TContext>(IHttpApplication<TContext> application)
+        {
+            _serverEvent.Set();
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.TestSites/Program.cs]: There's no guarantee Start will be called if there are other startup errors.

File: test/Microsoft.AspNetCore.Hosting.TestSites/project.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
+{
+  "version": "1.1.0-*",
+  "dependencies": {
+    "Microsoft.AspNetCore.Hosting": "1.1.0-*",
+    "Microsoft.AspNetCore.WebUtilities": "1.1.0-*",
+    "Microsoft.Extensions.Configuration": "1.1.0-*",
+    "Microsoft.Extensions.Configuration.CommandLine": "1.1.0-*",
+    "Microsoft.Extensions.Logging.Console": "1.1.0-*",
+    "Microsoft.Net.Http.Headers": "1.1.0-*"
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.TestSites/project.json]: not needed?

File: test/Microsoft.AspNetCore.Hosting.TestSites/project.json [hunk 8]
```diff
@@ -0,0 +1,36 @@
+{
+  "version": "1.1.0-*",
+  "dependencies": {
+    "Microsoft.AspNetCore.Hosting": "1.1.0-*",
+    "Microsoft.AspNetCore.WebUtilities": "1.1.0-*",
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.TestSites/project.json]: Not needed?

File: test/Microsoft.AspNetCore.Hosting.FunctionalTests/ShutdownTests.cs (hunk 4 above)
Comment (pakrym) [inline test/Microsoft.AspNetCore.Hosting.FunctionalTests/ShutdownTests.cs]: Delay is to make sure hosting started IServer.

File: test/Microsoft.AspNetCore.Hosting.TestSites/Program.cs (hunk 6 above)
Comment (pakrym) [inline test/Microsoft.AspNetCore.Hosting.TestSites/Program.cs]: Then process would be killed with timeout.

File: test/Microsoft.AspNetCore.Hosting.FunctionalTests/Microsoft.AspNetCore.Hosting.FunctionalTests.xproj (hunk 3 above)
Comment (dougbu) [inline test/Microsoft.AspNetCore.Hosting.FunctionalTests/Microsoft.AspNetCore.Hosting.FunctionalTests.xproj]: Shared version was out-of-date. I fixed that&hellip;

File: test/Microsoft.AspNetCore.Hosting.TestSites/project.json [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      "dependencies": {
+        "Microsoft.NETCore.App": {
+          "version": "1.1.0-*",
+          "type": "platform"
+        }
+      }
+    }
+  },
+  "tools": {
+    "Microsoft.AspNetCore.Server.IISIntegration.Tools": "1.0.0-preview2-final"
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.TestSites/project.json]: remove tools and scripts

File: test/Microsoft.AspNetCore.Hosting.TestSites/project.json [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    "Microsoft.Extensions.Configuration": "1.1.0-*",
+    "Microsoft.Extensions.Configuration.CommandLine": "1.1.0-*",
+    "Microsoft.Extensions.Logging.Console": "1.1.0-*",
+  },
+  "buildOptions": {
+    "emitEntryPoint": true
+  },
+  "publishOptions": {
+    "include": [
+      "web.config"
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Hosting.TestSites/project.json]: remove

**Comment by muratg** [comment]
@pranavkm FYI

**Pull request merged**